geomesa-kudu/geomesa-kudu-datastore/src/main/scala/org/locationtech/geomesa/kudu/data/KuduFeatureCollection.scala

```diff
@@ -0,0 +1,30 @@
+/***********************************************************************
+ * Copyright (c) 2013-2018 Commonwealth Computer Research, Inc.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Apache License, Version 2.0
+ * which accompanies this distribution and is available at
+ * http://www.opensource.org/licenses/apache2.0.php.
+ ***********************************************************************/
+
+package org.locationtech.geomesa.kudu.data
+
+import com.typesafe.scalalogging.LazyLogging
+import org.geotools.data.Query
+import org.locationtech.geomesa.index.geotools.{GeoMesaFeatureCollection, GeoMesaFeatureSource}
+import org.locationtech.geomesa.process.GeoMesaProcessVisitor
+import org.locationtech.geomesa.process.analytic.SamplingVisitor
+import org.opengis.feature.FeatureVisitor
+import org.opengis.util.ProgressListener
+
+class KuduFeatureCollection(source: GeoMesaFeatureSource, query: Query)
+    extends GeoMesaFeatureCollection(source, query) with LazyLogging {
+
+  override def accepts(visitor: FeatureVisitor, progress: ProgressListener): Unit =
+    visitor match {
+      case _: SamplingVisitor => super.accepts(visitor, progress) // TODO sampling not implemented yet
+      case v: GeoMesaProcessVisitor => v.execute(source, query)
```

Comment:
they aren't pushed down (since there isn't any push-down processing in kudu), but they are handled in the query planner implementation in an optimized manner. See https://github.com/elahrvivaz/geomesa/blob/fcr_kudu_ds/geomesa-kudu/geomesa-kudu-datastore/src/main/scala/org/locationtech/geomesa/kudu/index/KuduFeatureIndex.scala#L238